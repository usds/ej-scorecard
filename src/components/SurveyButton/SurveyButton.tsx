import React from 'react';
// import { Button } from '@trussworks/react-uswds';

import * as styles from './SurveyButton.module.scss';
import MainGridContainer from '../MainGridContainer';

// import { SurveyButtonProps } from '@/types';

const SurveyButton: React.FC = () => {
  return (
    <MainGridContainer className={styles.surveyButtonContainer}>
      <a
        href={`#`}
        target="_blank"
        rel="noreferrer"
        // Todo: Figure out how to get the external link working without having to use the icon
        // className={`usa-link usa-link--alt usa-link--external ${styles.surveyButton}`}
        className={styles.surveyButton}
      >
        {/* <Button type="button" className={styles.surveyButton}> */}
        <strong>Share your feedback</strong>
        {/* <img
            className={styles.launchIcon}
            src={launchIcon}
            alt={`launch icon`}
          /> */}
        {/* </Button> */}
      </a>
    </MainGridContainer>
  );
};

export default SurveyButton;
