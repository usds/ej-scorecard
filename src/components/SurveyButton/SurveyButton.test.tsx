import createComponentWithIntl from '@/test/testHelpers';

import SurveyButton from './SurveyButton';

describe(`rendering of SurveyButton Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(<SurveyButton />);

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
