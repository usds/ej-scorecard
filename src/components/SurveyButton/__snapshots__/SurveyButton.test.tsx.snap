// Vitest Snapshot v1

exports[`rendering of SurveyButton Component > checks if component renders 1`] = `
<div
  className="grid-container-desktop-lg"
  data-testid="gridContainer"
>
  <a
    href="https://eop.gov1.qualtrics.com/jfe/form/SV_aVPdC5qtPNhCw3s"
    rel="noreferrer"
    target="_blank"
  >
    <button
      className="usa-button"
      data-testid="button"
      type="button"
    >
      <strong>
        Share your feedback
      </strong>
      <img
        alt="launch icon"
        src="/node_modules/.pnpm/@uswds+uswds@3.3.0/node_modules/@uswds/uswds/dist/img/usa-icons/launch.svg"
      />
    </button>
  </a>
</div>
`;
